### PR TITLE
Label Component:  should have a padding of 8px on top and bottom

### LINKED ISSUE #1562: Labels should have 8px of padding not 5px
### Bug Report

- __Package version(s)__: (fill this out)
- __Browser and OS versions__: (fill this out if relevant)

#### Priorities and help requested (not applicable if asking question):

Are you willing to submit a PR to fix? (Yes, No)

Requested priority: (Blocking, High, Normal, Low) 

Products/sites affected: (if applicable)

### PR DESCRIPTION
**Pull request by aleksandrjPersonal**
#### Pull request checklist

- [X] Addresses an existing issue: Fixes #1562 
- [ ] Include a change request file if publishing 
- [ ] New feature, bugfix, or enhancement
  - [ ] Includes tests
- [ ] Documentation update

#### Description of changes

Adds more padding to Label

#### Focus areas to test
Label
(optional)

### CODE REVIEW

**Comment by mikewheaton** [comment]
As minor as this is, I think we still need a change file. Think I'd mark this as a 'patch' because it's essentially fixing a bug, where the Label didn't match the design previously. It's unlikely to be a breaking change for anyone.

**Comment by dzearing** [comment]
@mikewheaton I think this is a minor bump! CSS tweaks like this, which change the OUTER padding of a component can cause shifts in UX and therefore change an implicit contract. Which can be enormously unexpected. Are we sure this is correct? Have we tested this in products and scenarios? 

@gitjain @joschect Are the visual regression tests in the checkin workflow yet? Shouldn't this break a number of things?

**Comment by betrue-final-final** [comment]
There’s a bug where the spacing between labels and the tops of controls is too small (5px). This control seemed like the best place for a consistent spacing to be applied so it will work above all controls.

**Comment by aleksandrjPersonal** [comment]
Change file added as a 'minor'

**Comment by dzearing** [comment]
@betrue-final-final But that is probably a bad assumption.

For example, if it's the first component on the page, you would want it to have no spacing above it (other than the page's padding.)

You could add spacing below a component, but it's a really bad idea to assume every component should have some specific border.

What you want is a vertical stack of things, with inner padding in between them. Let the "things" be agnostic of the padding, let the stack maintain the inner padding.

**Comment by dzearing** [comment]
@betrue-final-final But he's changing it to have 8px of padding on both top and bottom of the label.

This seems wrong. Shouldn't it have no space above it? E.g. the first label on the page should have no additional padding above it beyond the container's inner padding?

**Comment by betrue-final-final** [comment]
Ok, I'll let you all engineer the thing, but can we make sure to fix the spacing between the text and labels of dropdowns and text fields so that they're 8px and not 5px? We figured that if we fixed it in one spot that we won't have to fuss with every single control. I don't know of an instance where it isn't 8px from the label to the control.

________________________________

**Comment by aleksandrjPersonal** [comment]
@dzearing Should I remove the top padding? Ok keep it? Previously it was 5px, so I thought top padding would be ok to modify. What do you recommend?

**Comment by dzearing** [comment]
@aleksandrjPersonal I would recommend that;

1. Labels don't have margins at all. Or second best would be to not change it.

2. Textboxes, Dropdowns, and any other components which render labels above them have a standardized padding in between the Label they render and the box boundaries.

I think we have made mistakes adding unnecessary padding around components at all; it might be acceptable in form elements to have bottom padding, but even that's an assumption. We removed padding rules around buttons altogether because there are different requirements in differnet scenarios.

File: packages/office-ui-fabric-react/src/components/Label/Label.scss [hunk 1]
```diff
@@ -15,7 +15,7 @@
   color: $ms-color-neutralPrimary;
   box-sizing: border-box;
   display: block;
-  padding: 5px 0;
+  padding: 8px 0;
```
Comment (dzearing) [inline packages/office-ui-fabric-react/src/components/Label/Label.scss]: Avoid this change, update individual components and ensure the padding is correct there.

Honestly, a better change would be to be consistent. Other components don't have `top-padding` but they do have `bottom-margin`. Why is Label different?

**Comment by betrue-final-final** [comment]
How do we ensure that padding is applied properly across elements? It's a given for most if not all controls that the label is 8px above the control. I feel like if we have every engineer set their own that it's going to be a bug pump. We should consult with @Angelaallison

File: packages/office-ui-fabric-react/src/components/Label/Label.scss (hunk 1 above)
Comment (aleksandrjPersonal) [inline packages/office-ui-fabric-react/src/components/Label/Label.scss]: @dzearing  Maybe we can leave the top padding at 5px and just change the bottom to 8px? Or maybe just close this with a won't fix?

Comment (aleksandrjPersonal) [inline packages/office-ui-fabric-react/src/components/Label/Label.scss]: @dzearing Should we remove the padding here and then deal with the regressions as they are discovered? Do we really want to comb through the entire project and look for Labels embedded in components and remove all their inherent padding and swap out for custom padding per example. How are we going to communicate to the Developers who use this framework that they need to implement their own 8px of padding when they use Label component? @betrue-final-final Ben? can we abandon this request? I wasn't expecting this level of push back on such a simple tweak...

Comment (betrue-final-final) [inline packages/office-ui-fabric-react/src/components/Label/Label.scss]: We should talk to @dzearing in real life.

**Comment by micahgodbolt** [comment]
Can we see an example in product where the padding is incorrect? This way we can verify if our changes are appropriate or if there is a better solution. (i.e. maybe it's the component using the label that should change).

**Comment by aleksandrjPersonal** [comment]
@micahgodbolt Here is where the problem was first discovered and flagged:
https://dev.office.com/fabric#/components/dropdown

**Comment by betrue-final-final** [comment]
After further investigation, this is actually by design. The redline was measuring from the baseline of the text to the control. Subtract the descender and you get 5px. This is by design.
image

**Pull request closed without merging**